checks/docker/prevent_sys_workdir_mounts.rego

```diff
@@ -0,0 +1,40 @@
+# METADATA
+# title: "WORKDIR should not be mounted on system dirs"
+# description: "WORKDIR should not be mounted on system directories to avoid container breakouts"
+# scope: package
+# schemas:
+# - input: schema["dockerfile"]
+# related_resources:
+# - https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#workdir
+# custom:
+#   id: DS030
+#   avd_id: AVD-DS-0030
+#   severity: HIGH
+#   short_code: avoid-sys-workdir-mounts
+#   recommended_action: "Avoid using system directories to mount WORKDIR"
+#   input:
+#     selector:
+#     - type: dockerfile
+package builtin.dockerfile.DS030
+
+import data.lib.docker
+
+sysdirs := {"/proc", "/boot", "/dev", "/initrd", "/lost+found"}
```

Comment:
best to add a tailing slash to prevent partial matching. for example someone might set workdir to `/development` which is legit but will be flagged by this check as it `startswith` `/dev`

Reply:
oh yeah good point, I've updated that and added a test.